File: src/main/java/artim/nemuos/karm/model/WorkItem.java
package artim.nemuos.karm.model;

public class WorkItem {
    private String workItemId;
    private String workItemTitle;
    private String workItemDescription;
    private String status;
    private String createdOn;
    private String lastModifiedOn;

    public String getWorkItemId() {
        return workItemId;
    }

    public void setWorkItemId(String workItemId) {
        this.workItemId = workItemId;
    }

    public String getWorkItemTitle() {
        return workItemTitle;
    }

    public void setWorkItemTitle(String workItemTitle) {
        this.workItemTitle = workItemTitle;
    }

    public String getWorkItemDescription() {
        return workItemDescription;
    }

    public void setWorkItemDescription(String workItemDescription) {
        this.workItemDescription = workItemDescription;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getCreatedOn() {
        return createdOn;
    }

    public void setCreatedOn(String createdOn) {
        this.createdOn = createdOn;
    }

    public String getLastModifiedOn() {
        return lastModifiedOn;
    }

    public void setLastModifiedOn(String lastModifiedOn) {
        this.lastModifiedOn = lastModifiedOn;
    }

    public WorkItem(String workItemId, String workItemTitle, String workItemDescription, String status, String createdOn, String lastModifiedOn) {
        this.workItemId = workItemId;
        this.workItemTitle = workItemTitle;
        this.workItemDescription = workItemDescription;
        this.status = status;
        this.createdOn = createdOn;
        this.lastModifiedOn = lastModifiedOn;
    }

    public WorkItem(String workItemId, String workItemTitle, String workItemDescription) {
        this.workItemId = workItemId;
        this.workItemTitle = workItemTitle;
        this.workItemDescription = workItemDescription;
        this.createdOn = java.time.Instant.now().toString();
        this.lastModifiedOn = java.time.Instant.now().toString();
        this.status = "TO-DO";
    }
public WorkItem(){

}
}

File: karm-android/app/src/main/java/com/nemuos/karm/model/WorkItem.kt
package com.nemuos.karm.model

import com.google.gson.annotations.SerializedName
import java.time.Instant

data class WorkItem(
    @SerializedName("workItemId")
    val workItemId: String = "",

    @SerializedName("workItemTitle")
    var workItemTitle: String = "",

    @SerializedName("workItemDescription")
    var workItemDescription: String = "",

    @SerializedName("status")
    var status: String = Status.TODO.displayName,

    @SerializedName("createdOn")
    val createdOn: String = Instant.now().toString(),

    @SerializedName("lastModifiedOn")
    var lastModifiedOn: String = Instant.now().toString(),

    @SerializedName("comments")
    val comments: MutableList<String> = mutableListOf()
) {
    fun addComment(comment: String) {
        val timestamp = Instant.now().toString().substring(0, 10)
        comments.add("$timestamp -> $comment")
        lastModifiedOn = Instant.now().toString()
    }

    fun getStatusEnum(): Status = Status.fromString(status)

    fun updateStatus(newStatus: Status) {
        status = newStatus.displayName
        lastModifiedOn = Instant.now().toString()
    }

    fun getFormattedCreatedDate(): String {
        return try {
            createdOn.substring(0, 10)
        } catch (e: Exception) {
            createdOn
        }
    }

    fun getFormattedLastModifiedDate(): String {
        return try {
            lastModifiedOn.substring(0, 10)
        } catch (e: Exception) {
            lastModifiedOn
        }
    }
}
